server/controllers/LikeController.js

```diff
@@ -0,0 +1,162 @@
+import db from '../database/models';
+import helper from '../utils/helpers/likeControllerHelper';
+
+const { Article, ArticleLike, User } = db;
+
+/**
+ * This class contains static methods for liking and unliking
+ * articles in the applications. It also contains methods
+ * for retrieving like informations for a particular article
+ * @class LikeController
+ * @returns {object} when called with the  new it returns an
+ * instance of LikeController but this is never used
+ */
+class LikeController {
+  /**
+   * This function is a middleware and is called when the user wants
+   * to like a particular article.
+   * @param {*} req the request object containing the user's request
+   * @param {*} res the response object used to send a respons to the user
+   * @param {*} next called when an error occurs
+   * @returns {void} does not return a value
+   * @author Chidiebere Oguejiofor
+   */
+  static likeArticle(req, res, next) {
+    const { slug } = req.params;
+    const { decoded: user } = req.body;
+
+    Article.findOne({
+      where: { slug },
+      attributes: ['id', 'title', 'slug'],
+      include: {
+        model: User,
+        as: 'author',
+        attributes: ['id', 'firstName', 'lastName', 'image'],
+      },
+    }).then((article) => {
+      if (article) {
+        const likeObj = {
+          userId: user.id,
+          articleId: article.id,
+        };
+        ArticleLike.findOrCreate({
+          where: likeObj,
+          defaults: likeObj,
+          attributes: [['updatedAt', 'timeLiked']],
+        }).then(([, created]) => {
+          res.status(created ? 201 : 200).json({
```

Comment:
separate this logic check from where you are using it, store it in a variable to make your code more readable. The same thing goes for line 51 and 52 below.